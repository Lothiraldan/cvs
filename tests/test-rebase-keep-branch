#!/bin/sh

echo "[extensions]" >> $HGRCPATH
echo "graphlog=" >> $HGRCPATH
echo "rebase=" >> $HGRCPATH

addcommit () {
    echo $1 > $1
    hg add $1
    hg commit -d "${2} 0" -m $1
}

hg init a
cd a
addcommit "c1" 0
addcommit "c2" 1

addcommit "l1" 2
addcommit "l2" 3

hg update -C 1
hg branch 'notdefault'
addcommit "r1" 4
hg glog --template '{rev}:{desc}:{branches}\n'

echo
echo '% Rebase a branch while preserving the branch name'
hg update -C 3
hg rebase -b 4 -d 3 --keepbranches 2>&1 | sed 's/\(saving bundle to \).*/\1/'
hg glog --template '{rev}:{desc}:{branches}\n'
