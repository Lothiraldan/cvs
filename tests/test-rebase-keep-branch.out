0 files updated, 0 files merged, 2 files removed, 0 files unresolved
marked working directory as branch notdefault
@  4:r1:notdefault
|
| o  3:l2:
| |
| o  2:l1:
|/
o  1:c2:
|
o  0:c1:


% Rebase a branch while preserving the branch name
2 files updated, 0 files merged, 1 files removed, 0 files unresolved
adding branch
adding changesets
adding manifests
adding file changes
added 1 changesets with 1 changes to 1 files
rebase completed
@  4:r1:notdefault
|
o  3:l2:
|
o  2:l1:
|
o  1:c2:
|
o  0:c1:

% dirstate branch should be "notdefault"
notdefault
